ethereumj-core/src/main/java/org/ethereum/sync/PeersPool.java

```diff
@@ -193,7 +194,13 @@ public void connect(Node node) {
         }
 
         synchronized (pendingConnections) {
-            ethereum.connect(node);
```

Comment:
Hi Nashatyrev,

I tried to solve a dependency cycle ( Ethereum -> WorldManager -> SyncManager -> PeersPool -> Ethereum ). This was the most non-intrusive way I thought of doing that, at the cost of small code duplication. 

To avoid code duplication entirely, I thought about having the PeersPool handle peer client connections (to me PeersPool should have all the functionality for connecting to a peer inside it instead of calling the top class, otherwise I don't see the point of having a connect method on it) and have PeersPool as a dependency of either EthereumImpl or WorldManager.

My understanding of how all these blocks interact and work with each other is still somewhat limited, so I'm trying to make as few changes as possible in each commit, so that it is easier to review and spot if my understandings of things take me on a wrong road.